src/Services/ItemUpdateService.php

```diff
@@ -288,91 +297,108 @@ public function generateContent()
+			$content[Client::VARIANT_ART_NO] = $sku;
+			$this->endpoint = Client::EDIT_PRODUCT_VARIANT;
 		}
-		elseif($variation['data']['variation']['isMain'] === true && count($attributeValue) == 0)
+		elseif(!is_null($itemLevel) && strlen($itemLevel) && $itemLevel == Client::EDIT_PRODUCT)
 		{
-			return Client::EDIT_PRODUCT;
+			$content[Client::PRODUCT_ART_NO] = $sku;
+			$this->endpoint = Client::EDIT_PRODUCT;
 		}
 		else
 		{
-			return Client::EDIT_PRODUCT;
+			//Todo add log
+			$this->getLogger(__METHOD__)->addReference('variationId', $variation->id)->error('ElasticExportRakutenDE::log.missingEndpoint');
+			return null;
 		}
-	}
 
-	/**
-	 * @param $variation
-	 * @return array
-	 */
-	private function getAttributeValueSetShortFrontendName($variation):array
-	{
-		$values = [];
-		$unsortedValues = [];
-
-		if(isset($variation['data']['attributes'][0]['attributeValueSetId'])
-			&& !is_null($variation['data']['attributes'][0]['attributeValueSetId']))
+		if($this->stockUpdate == "true" && $stillActive === true)
 		{
-			$i = 0;
-
-			if(isset($variation['data']['attributes']))
+			$stockList = $this->stockHelper->getStockList($variation);
+			if(count($stockList))
 			{
-				foreach($variation['data']['attributes'] as $attribute)
+				$content['stock'] = $stockList['stock'];
+				$content['available'] = false;
+
+				if($stockList['stock'] > 0)
 				{
-					$attributeValueName = '';
+					$content['available'] = true;
+				}
 
-					if(isset($attribute['attributeId']) && isset($attribute['valueId']))
+				if($this->endpoint == Client::EDIT_PRODUCT)
+				{
+					if($stockList['inventoryManagementActive'] == 1)
 					{
-						$attributeValueName = $this->marketAttributeHelperRepositoryContract->getAttributeValueName(
-							$attribute['attributeId'],
-							$attribute['valueId'],
-							'de');
+						$content['stock_policy'] = true;
 					}
-
-					if(strlen($attributeValueName) > 0)
+					else
 					{
-						$unsortedValues[$attribute['attributeId']] = $attributeValueName;
-						$i++;
+						$content['stock_policy'] = false;
 					}
 				}
+
+				if($stockList['updatedAt'] > strtotime($variation['data']['skus'][0]['exportedAt']))
+				{
+					$this->transferData = true;
+				}
 			}
+		}
+		elseif($this->stockUpdate == "true" && $stillActive === false)
+		{
+			$content['available'] = false;
+			$content['stock'] = 0;
+			$this->transferData = true;
+		}
 
-			$values = $unsortedValues;
+		if($this->priceUpdate == "true" && $stillActive === true)
+		{
+			$priceResponse = $this->priceHelper->getPrice($variation, $settings);
+			if($priceResponse instanceof SalesPriceSearchResponse)
+			{
+				if($priceResponse->price > 0)
+				{
+					$price = number_format((float)$priceResponse->price, 2, '.', '');
+				}
+				else
+				{
+					$price = '';
+				}
+
+				if($priceResponse->updatedAt > strtotime($variation['data']['skus'][0]['exportedAt']))
+				{
+					$this->transferData = true;
+				}
+
+				$content['price'] = (float)$price;
```

Comment:
on rakuten doku, it seems that they request a float to be sent, but it gives `error code 41` when is set as float